backend/src/main/java/com/htecgroup/skynest/controller/FileController.java

```diff
@@ -366,4 +366,95 @@ public ResponseEntity<FileResponse> getFileDetails(@PathVariable(name = "uuid")
+                  examples = {
+                    @ExampleObject(
+                        value =
+                            "{\"messages\":[\"Invalid session token\"],"
+                                + " \"status\": \"401\","
+                                + " \"timestamp\": \"2022-06-07 16:18:12\"}")
+                  })
+            }),
+        @ApiResponse(
+            responseCode = "403",
+            description = "Access denied",
+            content = {
+              @Content(
+                  mediaType = MediaType.APPLICATION_JSON_VALUE,
+                  schema = @Schema(implementation = ErrorMessage.class),
+                  examples = {
+                    @ExampleObject(
+                        value =
+                            "{\"messages\":[\"User does not have access to bucket\"],"
+                                + " \"status\": \"403\","
+                                + " \"timestamp\": \"2022-06-07 16:18:12\"}")
+                  })
+            }),
+        @ApiResponse(
+            responseCode = "404",
+            description = "File not found",
+            content = {
+              @Content(
+                  mediaType = MediaType.APPLICATION_JSON_VALUE,
+                  schema = @Schema(implementation = ErrorMessage.class),
+                  examples = {
+                    @ExampleObject(
+                        value =
+                            "{\"messages\":[\"File not found\"],"
+                                + " \"status\": \"404\","
+                                + " \"timestamp\": \"2022-06-03 16:18:12\"}"),
+                  })
+            }),
+        @ApiResponse(
+            responseCode = "409",
+            description = "File already deleted",
+            content = {
+              @Content(
+                  mediaType = MediaType.APPLICATION_JSON_VALUE,
+                  schema = @Schema(implementation = ErrorMessage.class),
+                  examples = {
+                    @ExampleObject(
+                        value =
+                            "{\"messages\":[\"File already deleted\"],"
+                                + " \"status\": \"404\","
+                                + " \"timestamp\": \"2022-06-03 16:18:12\"}"),
+                  })
+            }),
+        @ApiResponse(
+            responseCode = "500",
+            description = "Internal Server Error",
+            content = {
+              @Content(
+                  mediaType = MediaType.APPLICATION_JSON_VALUE,
+                  schema = @Schema(implementation = String.class),
+                  examples = {@ExampleObject(value = "Internal Server Error")})
+            })
+      })
+  @DeleteMapping("/{fileId}")
+  public ResponseEntity<Boolean> deleteFile(@PathVariable UUID fileId) {
+
+    FileResponse fileResponse = fileService.deleteFile(fileId);
+    log.info("File {} successfully deleted", fileResponse.getName());
```

Comment:
record this as a delete action in the service method

Reply:
Totally forgot, thanks for noticing

Reply:
I have one a little bit advanced way for you to do it.. 
You can create a Custom annotation service layer for delete action and provide the specific input params there.
That custom annotation would be handled by AOP (Aspect oriented programming - module provided by Spring) and that Aspect would be triggered after the result of the service is executed. 

Don't have to do it now, but investigate it and I believe you have never used that approach and you will find it interesting and maybe it will give you different perspectives :)